tests/kafkatest/services/verifiable_producer.py

```diff
@@ -147,9 +147,10 @@ def _worker(self, idx, node):
         if self.enable_idempotence:
             self.logger.info("Setting up an idempotent producer")
             producer_prop_file += "\nmax.in.flight.requests.per.connection=5\n"
-            producer_prop_file += "\nretries=1000000\n"
             producer_prop_file += "\nenable.idempotence=true\n"
-        elif self.retries is not None:
+            if self.retries is None:
+                producer_prop_file += "\nretries=1000000\n"
+        if self.retries is not None:
```

Comment:
In retries document https://kafka.apache.org/documentation/#producerconfigs_retries, it mentions
```
Enabling idempotence requires this config value to be greater than 0.
If conflicting configurations are set and idempotence is not explicitly enabled, idempotence is disabled.
```

I think we should add assertion to `__int__` to make sure if enable_idempotence is true, then retries should not be zero.